Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1223 @@
+                return false;
+            }
+        }
+    });
+
+    var BILLBOARD_SIZE = 32;
+
+    function isZipFile(blob) {
+        var magicBlob = blob.slice(0, Math.min(4, blob.size));
+        var deferred = when.defer();
+        var reader = new FileReader();
+        reader.addEventListener('load', function() {
+            deferred.resolve(new DataView(reader.result).getUint32(0, false) === 0x504b0304);
+        });
+        reader.addEventListener('error', function() {
+            deferred.reject(reader.error);
+        });
+        reader.readAsArrayBuffer(magicBlob);
+        return deferred;
+    }
+
+    function readBlobAsText(blob) {
+        var deferred = when.defer();
+        var reader = new FileReader();
+        reader.addEventListener('load', function() {
+            deferred.resolve(reader.result);
+        });
+        reader.addEventListener('error', function() {
+            deferred.reject(reader.error);
+        });
+        reader.readAsText(blob);
+        return deferred;
+    }
+
+    function loadXmlFromZip(reader, entry, uriResolver, deferred) {
+        entry.getData(new zip.TextWriter(), function(text) {
+            uriResolver.gpx = parser.parseFromString(text, 'application/xml');
+            deferred.resolve();
+        });
+    }
+
+    function proxyUrl(url, proxy) {
+        if (defined(proxy)) {
+            if (new Uri(url).isAbsolute()) {
+                url = proxy.getURL(url);
+            }
+        }
+        return url;
+    }
+
+    function getOrCreateEntity(node, entityCollection) {
+        var id = queryStringAttribute(node, 'id');
+        id = defined(id) ? id : createGuid();
+        var entity = entityCollection.getOrCreateEntity(id);
+        return entity;
+    }
+
+    function readCoordinate(value) {
+        if (!defined(value)) {
+            return undefined;
+        }
+
+        var digits = value.match(/[^\s,\n]+/g);
+        if (digits.length !== 2 && digits.length !== 3) {
+            window.console.log('GPX - Invalid coordinates: ' + value);
+            return undefined;
+        }
+
+        var longitude = parseFloat(digits[0]);
+        var latitude = parseFloat(digits[1]);
+        var height = parseFloat(digits[2]);
+
+        longitude = isNaN(longitude) ? 0.0 : longitude;
+        latitude = isNaN(latitude) ? 0.0 : latitude;
+        height = isNaN(height) ? 0.0 : height;
+
+        return Cartesian3.fromDegrees(longitude, latitude, height);
+    }
+
+    function readCoordinates(element) {
+        if (!defined(element)) {
+            return undefined;
+        }
+
+        var tuples = element.textContent.match(/[^\s\n]+/g);
+        var length = tuples.length;
+        var result = new Array(length);
+        var resultIndex = 0;
+        for (var i = 0; i < length; i++) {
+            result[resultIndex++] = readCoordinate(tuples[i]);
+        }
+        return result;
+    }
+
+    function getCoordinatesString(node) {
```

Comment:
We should have this function return a `Cartesian3` instead of a string, this will make processing time faster and also allow you to use elevation.  Just use `Cartesian3.fromDegrees`